### PR TITLE
Add `to_chainer_device` and `LinkAsTorchModel.to(device)`

### PR DESCRIPTION
**Pull request by asi1024**
chainer device is used for operating `model.to_device(...)` before converting the model with `cpm.LinkAsTorchModel`.

### CODE REVIEW

**Comment by niboshi** [comment]
How about overriding `cpm.LinkAsTorchModel.to()` instead?

**Comment by asi1024** [comment]
I want to convert before passing to `cpm.LinkAsTorchModel` to operate the first forward for the initialization.

**Comment by asi1024** [comment]
I want to use `chainer_device.use()` too.

**Comment by niboshi** [comment]
>I want to convert before passing to `cpm.LinkAsTorchModel` to operate the first forward for the initialization.

I think you can feed a dummy batch on CPU.

**Comment by asi1024** [comment]
@niboshi PTAL.

**Comment by niboshi** [comment]
How about separating `to_chainer_device` in another PR and implement the reverse conversion (PyTorch -> Chainer device) as well?

File: chainer_pytorch_migration/parameter.py [hunk 1]
```diff
@@ -70,6 +73,12 @@ def __as_tensor(self, value):
             return _ChainerTensor(value)
         return value
 
+    def to(self, *, device):
```
Comment (niboshi) [inline chainer_pytorch_migration/parameter.py]: `to` can accept other things than devices, such as dtypes.

File: chainer_pytorch_migration/parameter.py [hunk 2]
```diff
@@ -29,8 +29,9 @@ class LinkAsTorchModel(torch.nn.Module):
         link (:class:`chainer.Link`): A link. Must have been initialized.
     '''
 
-    def __init__(self, link):
+    def __init__(self, link, **kwargs):
         super().__init__()
+        device = kwargs.pop('_device', None)
```
Comment (niboshi) [inline chainer_pytorch_migration/parameter.py]: Why is this needed, if `to(device)` is supported?

File: chainer_pytorch_migration/parameter.py [hunk 3]
```diff
@@ -70,6 +73,12 @@ def __as_tensor(self, value):
             return _ChainerTensor(value)
         return value
 
+    def to(self, *, device):
+        if device is None:
+            return self
+        device = cpm.to_chainer_device(torch.device(device))
+        return LinkAsTorchModel(self.link, _device=device)
```
Comment (niboshi) [inline chainer_pytorch_migration/parameter.py]: `to` should modify and return `self`, instead of creating a new instance.

Comment (asi1024) [reply chainer_pytorch_migration/parameter.py]: I'm trying to make a fix for this comment, but it seems that it will not be resolved soon. So, I dropped the commit for `LinkAsTorchModel.to` and will support it in another PR.

Comment (niboshi) [reply chainer_pytorch_migration/parameter.py]: Then, how about implementing the reverse conversion for completeness?

Comment (asi1024) [reply chainer_pytorch_migration/parameter.py]: @niboshi Added `to_torch_device`. PTAL.

**Comment by niboshi** [comment]
LGTM
/test
